source/extensions/filters/http/compressor/compressor_filter.cc

```diff
@@ -604,6 +606,28 @@ void CompressorFilter::sanitizeEtagHeader(Http::ResponseHeaderMap& headers) {
   }
 }
 
+// True if response compression is enabled.
+bool CompressorFilter::compressionEnabled(
+    const CompressorFilterConfig::ResponseDirectionConfig& config) const {
+  const CompressorPerRouteFilterConfig* per_route_config =
+      Http::Utility::resolveMostSpecificPerFilterConfig<CompressorPerRouteFilterConfig>(
+          decoder_callbacks_);
+  if (per_route_config) {
+    switch (per_route_config->config().override_case()) {
+    case CompressorPerRoute::kDisabled:
+      return false;
+    case CompressorPerRoute::kOverrides:
+      if (per_route_config->config().overrides().has_response_direction_config()) {
+        return true;
+      }
+      break;
+    case CompressorPerRoute::OVERRIDE_NOT_SET:
+      ASSERT(false, "CompressorPerRoute::override not set. Ignoring.");
```

Comment:
To continue the comment below on config validation, IIUC the `compressionEnabled` is called on the data-plane path, and so the validation cannot be done here. If the validation was done in the config-plane path it could be rejected/acted upon once there.

Reply:
Yes, `compressionEnabled` is data-plane, but the validation is done elsewhere. I added a comment to document that, and a config_test.cc test to confirm it.

Reply:
Thanks for adding the config test. +1 to `PANIC_DUE_TO_CORRUPT_ENUM` since this should not happen normally.

Reply:
Switched to PANIC_DUE_TO_CORRUPT_ENUM.

Reply:
Converted per-route proto to optional<bool>.